### ISSUE TITLE
Parse non-ASCII identifiers

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#64]
JavaScript allows certain non-ASCII characters in identifiers. For example:

```javascript
let Ƞ = "hello";
console.log(Ƞ);  // Prints 'hello'.
```

This confuses quick-lint-js. Make our lexer parse UTF-8 properly and recognize identifier characters.

**Comment by strager** [comment]
I didn't thoroughly test it, but this feature should be implemented in commit 7b945fd8a7cb3e40e73b4f5fe39bb5b69be8d0b8.
